FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_light_weight/DemoTasks/LightWeightMQTTExample.c

```diff
@@ -41,41 +54,938 @@
 #include "FreeRTOS.h"
 #include "task.h"
 
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/* MQTT library includes. */
+#include "mqtt_lightweight.h"
+
 /*-----------------------------------------------------------*/
 
+/* Compile time error for undefined configs. */
+#ifndef democonfigMQTT_BROKER_ENDPOINT
+    #error "Define the config democonfigMQTT_BROKER_ENDPOINT by following the instructions in file demo_config.h."
+#endif
+
+/*-----------------------------------------------------------*/
+
+/* Default values for configs. */
+#ifndef democonfigCLIENT_IDENTIFIER
+
 /**
- * @brief The task used to demonstrate the light weight MQTT API.
+ * @brief The MQTT client identifier used in this example.  Each client identifier
+ * must be unique, so edit as required to ensure no two clients connecting to the
+ * same broker use the same client identifier.
+ *
+ * @note Appending __TIME__ to the client id string will reduce the possibility of a
+ * client id collision in the broker. Note that the appended time is the compilation
```

Comment:
Why not use a random number instead to avoid the compile time macro problem?

Reply:
Didn't use a random number here since we are recommending to use a unique identifier by editing the config. __TIME__ is added as a way to generate the unique identifier out of the box for the default config. If we are suggesting an edit to the config to make it unique, a random number appending may not be useful for the end customer.